### PR TITLE
Add support for outputting environment files.

### PR DESCRIPTION
**Pull request by matthewwardrop**
Since it was straightforward to do, this is a quick patch that achieves the desired results of #97 . Happy to take feedback and modify this patch, to hand it over to the primary devs, or discard it (hopefully it doesn't come to this :).

With this current patch you can use it like this:
```
$ conda-lock --mamba -f conda_environment.yml -p linux-64 -p osx-64 -k env -k explicit --filename-template "lockfiles/conda-{platform}.lock"
generating lockfile for linux-64
Install lock using: conda env create --name YOURENV --file lockfiles/conda-linux-64.lock.yml
Install lock using: conda create --name YOURENV --file lockfiles/conda-linux-64.lock
generating lockfile for osx-64
Install lock using: conda env create --name YOURENV --file lockfiles/conda-osx-64.lock.yml
Install lock using: conda create --name YOURENV --file lockfiles/conda-osx-64.lock
```

This is now working nicely for my use-cases, and I hope you will consider it for merging!

### CODE REVIEW

**Comment by mariusvniekerk** [comment]
So the large drawback of this approach is that installing this environment will still invoke the conda solver and won't neccesarily result in the solve present in that environment unfortunately.

This can be caused by repodata being patched to correct dependencies (defaults, conda-forge, bioconda and some of the other larger channels do this).

**Comment by matthewwardrop** [comment]
@mariusvniekerk thanks for taking the time to review this PR.

Aye... Invoking the solver means it is possible that the resulting environment won't be exactly the same.... But it should be close. In our case we have a Conda environment that needs to be shipped into production, for which we use explicit locks, and then we have a version of the locks in an environment file which can be used iteratively in a user's environment to bring it in line with our upstream environment, which the user is free to extend and mutate. In the past we blindly applied the locks, but that was much less robust, and frequently had to be reset... Which was annoying and slow for our users.

Are you willing to have support for generating these enriched environment files upstream? Or is it too antithetical to the goals of this project?

**Comment by mariusvniekerk** [comment]
I think we can support them just with lots of warnings to users telling about their shortcomings

**Comment by matthewwardrop** [comment]
@mariusvniekerk I've patched the black code formatting issue, and added some warnings. If the patch otherwise looks okay to you, I can add a environment lock generation test if that would be helpful.

**Comment by mariusvniekerk** [comment]
Yeah this definitely would need some test to ensure that future changes don't break this feature

**Comment by mariusvniekerk** [comment]
In order to not have to fight black (and the other precommit parts) i'd advise that you install [pre-commit](https://pre-commit.com/) and run `pre-commit init` in your checkout of the repo.

File: conda_lock/conda_lock.py [hunk 1]
```diff
@@ -416,56 +432,88 @@ def create_lockfile_from_spec(
...
+        raise ValueError(f"Unrecognised lock kind {kind}.")
 
+    logging.debug("lockfile_contents:\n%s\n", lockfile_contents)
     return lockfile_contents
 
 
+def create_environment_file_from_spec(
+    *, channels: Sequence[str], conda: PathLike, spec: LockSpecification
+) -> List[str]:
+    ...
```
[the remark this answers is not in the document]
Comment (matthewwardrop) [reply conda_lock/conda_lock.py:526]: Apologies, no.

**Comment by matthewwardrop** [comment]
Okay, I've extended the unit tests, and removed the vestigial code. Let me know if there's anything else you'd like to have done. Also, feel free to push directly to this branch if you want to :).

**Comment by matthewwardrop** [comment]
Thanks @mariusvniekerk ! (And sorry re:linting issues; the instructions for pre-commit didn't work for me, and I forgot to lint the tests also).

**Pull request merged**